apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx

```diff
@@ -140,47 +154,87 @@ const Body = ({ data, onClose }: { data: Data; onClose?: () => void | undefined
     [dispatch, exchange, transactionParams, provider],
   );
 
-  const onBroadcastSuccess = useCallback(
-    (operation: Operation) => {
-      // If swap we save to swap history and keep open the drawer
-      if (swapId && toAccount && magnitudeAwareRate && sourceCurrency && targetCurrency) {
-        const newResult = {
-          operation,
-          swapId,
-        };
-        updateAccount({
-          result: newResult,
-          magnitudeAwareRate,
-        });
-        setResult({
+  const getResultByTransactionType = (
+    isSwapTransaction: "" | CryptoCurrency | TokenCurrency | null | undefined,
+  ) => {
+    return isSwapTransaction
+      ? {
           swapId,
+          mode: ExchangeModeEnum.Swap,
           provider,
-          sourceCurrency,
-          targetCurrency,
-        });
-
-        if (getEnv("DISABLE_TRANSACTION_BROADCAST")) {
-          return onCancel(new DisabledTransactionBroadcastError());
+          sourceCurrency: sourceCurrency as Currency,
+          targetCurrency: targetCurrency as Currency,
         }
-        onResult(operation);
-        // else not swap i.e card and sell we close the drawer
+      : {
+          provider,
+          mode: ExchangeModeEnum.Sell,
+          sourceCurrency: sourceCurrency as Currency,
+        };
+  };
+
+  const handleTransactionResult = (result: ResultsState, operation: Operation) => {
+    setResult(result);
+
+    onResult(operation);
+  };
+
+  const handleSwapTransaction = (operation: Operation, result: ResultsState) => {
+    const newResult = {
+      operation,
+      swapId: swapId as string,
+    };
+
+    updateAccount({
+      result: newResult,
+      magnitudeAwareRate: magnitudeAwareRate as BigNumber,
+    });
+
+    handleTransactionResult(result, operation);
+  };
+
+  const handleSellTransaction = (operation: Operation, result: ResultsState) => {
+    handleTransactionResult(result, operation);
+  };
+
+  const onBroadcastSuccess = useCallback(
+    (operation: Operation) => {
+      if (getEnv("DISABLE_TRANSACTION_BROADCAST")) {
+        return onCancel(new DisabledTransactionBroadcastError());
+      }
+
+      const isSwapTransaction =
+        swapId && toAccount && magnitudeAwareRate && sourceCurrency && targetCurrency;
+
+      const isSellTransaction =
+        (data.exchangeType === ExchangeType.SELL || data.exchangeType === ExchangeType.SELL_NG) &&
+        sourceCurrency;
```

Comment:
I though you want to use:
```suggestion
      const isSellTransaction =
        (data.exchangeType === ExchangeType.SELL || data.exchangeType === ExchangeType.SELL_NG);
```

Reply:
`sourceCurrency` is just serving as an extra guarding clause, like the swap team does